### PR TITLE
Added config support for flapdoodle features

### PR DESCRIPTION
**Pull request by mgaceanu**
Specify in the config the flapdoodle features to be enabled.

### CODE REVIEW

File: src/main/java/com/github/joelittlejohn/embedmongo/AbstractEmbeddedMongoMojo.java [hunk 1]
```diff
@@ -102,19 +111,26 @@ protected IFeatureAwareVersion getVersion() {
             versionEnumName = "V" + versionEnumName;
         }
 
+
+        Feature[] features = new Feature[0];
+        try {
+            features = Stream.of(this.features.split(",")).map(String::trim).map(String::toUpperCase).map(Feature::valueOf).toArray(Feature[]::new);
```
Comment (codescale) [inline src/main/java/com/github/joelittlejohn/embedmongo/AbstractEmbeddedMongoMojo.java]: I'm not sure, but if the field `features` is not set and stays `null` won't this `this.features.split(",")` throw a NullPointer? Maybe `features` needs a default?

Comment (mgaceanu) [reply src/main/java/com/github/joelittlejohn/embedmongo/AbstractEmbeddedMongoMojo.java]: I added as suggested a null check, unit test and documentation.

**Comment by codescale** [comment]
Well done @mgaceanu . Thanks for the PR!

**Comment by joelittlejohn** [comment]
I'm happy to release 0.4.2 now to get this out.

**Comment by mgaceanu** [comment]
@joelittlejohn it seems 0.4.2 is not yet released. Did you decided to delay it?

**Comment by rkaltreider** [comment]
> I'm happy to release 0.4.2 now to get this out.

when is thins coming out?  looking forward to it!

**Comment by joelittlejohn** [comment]
Sorry, I haven't had time to look at this yet. I'm travelling right now but back home at the weekend and I'll release.

**Comment by rkaltreider** [comment]
Thank you sir, greatly looking forward to it.

**Comment by rkaltreider** [comment]
Any luck with the release?

**Comment by joelittlejohn** [comment]
Thanks for the reminder. I've just released 0.4.2. It should usable right now I think, and visible in Maven central search in a few hours.

**Comment by rkaltreider** [comment]
Thanks! You’re the man! The new features work great…I can finally embed mongo 4.0.12!!! Of course now they released 4.2.0 which only runs on windows server 2012 so the paths need updated again, but that’s will be handled in the other project….may just need to update dependency in the embedmongo plugin.

**Comment by joelittlejohn** [comment]
Good stuff, thanks for the update! Whenever you need to bump the version of the flapdoodle library in this project, feel free to submit a PR and I'll try to get it merged and released more quickly :)

**Comment by rkaltreider** [comment]
Will do! Thanks again!

**Pull request merged**